--- a/local.mk.orig	2015-02-23 22:55:52.507612312 +0100
+++ b/local.mk	2015-02-23 22:57:06.491611875 +0100
@@ -277,7 +277,7 @@
 allhtml = $(dist_doc_DATA) $(dist_examples_DATA) $(dist_modules_DATA) $(dist_classes_DATA)
 
 $(allhtml): $(EXTRA_ext_posix_posix_la_SOURCES) $(ext_posix_posix_la_SOURCES)
-	test -d $(builddir)/doc || mkdir $(builddir)/doc
+	test -d $(builddir)/doc || mkdir -p $(builddir)/doc
 if HAVE_LDOC
 	$(LDOC) -c build-aux/config.ld -d $(abs_srcdir)/doc .
 else
